core/src/main/java/nl/nn/adapterframework/pipes/IteratingPipe.java

```diff
@@ -414,18 +414,34 @@ public void waitForResults() throws SenderException, IOException {
 				try {
 					guard.waitForAllResources();
 					int count = 0;
+					List<Throwable> exceptions = new ArrayList<>();
 					for (ParallelSenderExecutor pse : executorList) {
 						count++;
-						String itemResult;
+						String itemResult="";
 						if (pse.getThrowable() == null) {
 							SenderResult senderResult = pse.getReply();
 							if (senderResult.isSuccess()) {
 								itemResult = senderResult.getResult().asString();
-							} else {
-								itemResult = "<exception>"+ XmlEncodingUtils.encodeChars(senderResult.getResult().asString())+"</exception>";
+							} 
+							else {
+								if(isIgnoreExceptions()){
+									itemResult = "<exception>"+ XmlEncodingUtils.encodeChars(senderResult.getResult().asString())+"</exception>";
+								} else {
+									SenderException se = new SenderException(senderResult.getResult().asString());
+									exceptions.add(se);
+								}
 							}
 						} else {
-							itemResult = "<exception>"+ XmlEncodingUtils.encodeChars(pse.getThrowable().getMessage())+"</exception>";
+							if(isIgnoreExceptions()){
+								itemResult = "<exception>"+ XmlEncodingUtils.encodeChars(pse.getThrowable().getMessage())+"</exception>";
+							} else {
+								exceptions.add(pse.getThrowable());
+							}
+						}
+						if(!exceptions.isEmpty()){
+							SenderException se = new SenderException("Something went wrong in one of the parallel running pipes: ");
```

Comment:
It's not always a pipe that's being executed, and you typically never want the exception message to be left on a cliffhanger (using `: ` without the actual reason).

```suggestion
							SenderException se = new SenderException("an error occurred during parallel execution");
```